packages/botonic-api/src/rest/routes/users.ts

```diff
@@ -1,55 +1,182 @@
+import { User } from '@botonic/core/src/models/user'
 import { Router } from 'express'
-// import { dataProviderFactory } from '../../data-provider/'
-// import { ApiGatewayManagementApi } from 'aws-sdk'
+import { checkSchema, matchedData, validationResult } from 'express-validator'
+
+import { dataProviderFactory } from '../../data-provider'
+import {
+  getOptionalSchema,
+  limitParamSchema,
+  offsetParamSchema,
+} from './validation/common'
+import {
+  userIdParamSchema,
+  userSchema,
+  userWithUserIdParamSchema,
+} from './validation/users'
 
 const router = Router()
 
 router
   .route('/')
-  .get((_, res) => {
-    res.send('GET HTTP method on user resource')
-  })
-  .post((_, res) => {
-    res.send('POST HTTP method on user resource')
+  .get(
+    checkSchema({
+      limit: limitParamSchema,
+      offset: offsetParamSchema,
```

Comment:
The most common way to add pagination to an API is to define a param "page" (by default 1) and an optional "pageSize" (default typically 10 or 20), with pageSize having a maximum allowed (for instance 100).
Is there any particular reason you used "limit" and "offset"?

Reply:
I've seen it done both ways and decided to use the limit/offset one. 
For the developer maybe it's easier to use the page/pageSize...I'll change it to page/pageSize in another PR and I'll add also the `prev`, `next`, `count`, `results` in the api response as you said in another comment.